data-analysis/template.yaml

```diff
@@ -394,10 +394,10 @@ Resources:
             Action:
               - kms:*
             Resource: '*'
-          - Sid: Enable Decrypt access for Query Results account
+          - Sid: Enable Decrypt access for Query Results account generate download Lambda
             Effect: Allow
             Principal:
-              AWS: '{{resolve:ssm:QueryResultsAccountRootArn}}'
+              AWS: !Sub 'arn:aws:iam::{{resolve:ssm:QueryResultsAccountNumber}}:role/txma-query-results-build-generate-download-role'
```

Comment:
This won't deploy unless the role exists. Which we might not want to enforce as I'm guessing the role is created within an application stack. Since this is static infrastructure, this will most likely be deployed first.

Another way to do this would be to keep the RootArn param, and use conditions to do the restrictions.